### ISSUE TITLE
500 internal error results in blocked response

### ISSUE THREAD

**Issue by CitizenBeta** [fruitcake/laravel-cors#365]
Today I ran into this error:

> Access to XMLHttpRequest at 'https://domain.test/api/users' from origin 'http://localhost:8100' has been blocked by CORS policy: The 'Access-Control-Allow-Origin' header contains multiple values '*, http://localhost:8100', but only one is allowed.

Which turned out to be a 500 internal error that I was able to fix in my code. The problem is, 500 errors do happen out in the field and it would be great if Laravel was able to return that 500 error rather than being an unreadable response due to CORS. That way, my app can show a message to the user about that error.

What do I need to do with my CORS setup to return the error rather than having it get blocked?

The only thing I can think of is wrapping every single request in a try/catch and having the catch return a response deliberately rather than relying on the standard response generated by Laravel when an exception occurs. This seems very unlaravel-like and bulky. Is there a better way?

**Comment by FlorianB385** [comment]
I've the same problem but any solutions doesn't work for me… Anyone else a solution for this problem?

**Comment by barryvdh** [comment]
I'm closing this due to inactivity. If this issue is still relevant on the latest version (2.0.1 at the moment), please open a new issue with detailed information.
